libs/src/services/solanaWeb3Manager/tokenAccount.js

```diff
@@ -0,0 +1,149 @@
+const { ASSOCIATED_TOKEN_PROGRAM_ID, Token } = require('@solana/spl-token')
+const {
+  PublicKey,
+  SystemProgram,
+  SYSVAR_RENT_PUBKEY,
+  Keypair
+} = require('@solana/web3.js')
+
+/**
+ * Finds the associated token address given a solana wallet public key
+ * @param {PublicKey} solanaWalletKey Public Key for a given solana account (a wallet)
+ * @param {PublicKey} mintKey
+ * @param {PublicKey} solanaTokenProgramKey
+ * @returns {PublicKey} token account public key
+ */
+async function findAssociatedTokenAddress ({
+  solanaWalletKey,
+  mintKey,
+  solanaTokenProgramKey
+}) {
+  const addresses = await PublicKey.findProgramAddress(
+    [
+      solanaWalletKey.toBuffer(),
+      solanaTokenProgramKey.toBuffer(),
+      mintKey.toBuffer()
+    ],
+    ASSOCIATED_TOKEN_PROGRAM_ID
+  )
+  return addresses[0]
+}
+
+/**
+ * Gets token account information (e.g. balance, ownership, etc.)
+ * @param {PublicKey} tokenAccountAddressKey
+ * @param {PublicKey} mintKey
+ * @param {PublicKey} solanaTokenProgramKey
+ * @param {Connection} connection
+ * @returns {AccountInfo}
+ */
+async function getAssociatedTokenAccountInfo ({
+  tokenAccountAddressKey,
+  mintKey,
+  solanaTokenProgramKey,
+  connection
+}) {
+  const token = new Token(
+    connection,
+    mintKey,
+    solanaTokenProgramKey,
+    Keypair.generate()
+  )
+  const info = await token.getAccountInfo(tokenAccountAddressKey)
+  return info
+}
+
+/**
+ * Creates an associated token account for a given solana account (a wallet)
+ * @param {PublicKey} feePayerKey
+ * @param {PublicKey} solanaWalletKey the wallet we wish to create a token account for
+ * @param {PublicKey} mintKey
+ * @param {PublicKey} solanaTokenProgramKey
+ * @param {Connection} connection
+ * @param {IdentityService} identityService
+ */
+async function createAssociatedTokenAccount ({
+  feePayerKey,
+  solanaWalletKey,
+  mintKey,
+  solanaTokenProgramKey,
+  connection,
+  identityService
+}) {
+  const associatedTokenAddress = await findAssociatedTokenAddress({
+    solanaWalletKey,
+    mintKey,
+    solanaTokenProgramKey
+  })
+
+  const accounts = [
+    // 0. `[sw]` Funding account (must be a system account)
+    {
+      pubkey: feePayerKey,
```

Comment:
Do we want to be the rent payer on this non-user-bank solana token account?
In this case, I image this user has a third party solana wallet which (I imagine, but am just guessing) makes the token account for them? 
Or, the user is trying to send SPL to another user who has not set it up?
I'm not against being the fee payer, but just don't know if it's the behavior we want.

Reply:
yeah, that's super fair. tbh, im not totally sure. @piazzatron & @hareeshnagaraj  for thoughts? we *definitely* need to support this behavior of creating the wAudio token account for a system account that never interacted w/ wAudio

Reply:
is it plausible to gate the creation with some kind of additional check...? not within this function necessarily but perhaps  in the actual call within the client?

Reply:
not sure i follow -- what kind of check would you add?